### PR TITLE
Linting some code

### PR DESCRIPTION
**Pull request by ldez**
### What does this PR do?

Fixes some problems:
- remove dead code
- use valid pattern with `print-format`
- missing errors return
- positive error codes

Add a "raw" gometalinter configuration, for now it's not use anywhere.

### Motivation

Enhance code.

### More

- [x] Added/updated tests

### CODE REVIEW

File: cmd/traefik/traefik.go [hunk 1]
```diff
@@ -75,7 +75,7 @@ Complete documentation is available at https://traefik.io`,
 		os.Exit(-1)
 	}
 
-	if _, err := f.Parse(usedCmd); err != nil {
+	if _, err = f.Parse(usedCmd); err != nil {
 		if err == pflag.ErrHelp {
 			os.Exit(0)
 		}
```
Comment (timoreimann) [inline cmd/traefik/traefik.go]: Unrelated to this PR: I think line 83 should return an exit code of `1`; negative exit codes are at least very uncommon, if not possibly also troublesome.

File: cmd/traefik/traefik.go [hunk 2]
```diff
@@ -75,7 +75,7 @@ Complete documentation is available at https://traefik.io`,
 		os.Exit(-1)
 	}
 
-	if _, err := f.Parse(usedCmd); err != nil {
+	if _, err = f.Parse(usedCmd); err != nil {
```
[the remark this answers is not in the document]
Comment (vdemeester) [reply cmd/traefik/traefik.go]: same question as @timoreimann why removing `:=` ? 🤔

**Comment by ldez** [comment]
@timoreimann My changes in this PR should not be taken as a way forward.

If we want to have a better review analysis, we need linters.

The case-by-case approach consume a lot of time, and can leads to a philosophical approach (for me it's like the tab/space debate).

Linters doesn't allow to separate error shallow declaration from others, and I don't think it's a good idea to try to do that.

The error shallow declaration can lead to some weird edge case, I've already fixed it in the code base.
Your first comment is the proof: 2 variables with the same name but different scope, it does not facilitate the reading and can lead to weird or misunderstood behavior.

Linters are not perfect and we have to try to configure them as we need, but not by using fine exceptions (case-by-case approach).

We need global rules even if, in some cases, it's dumb.

For now, it's just some minor fixes, we will not use more linters, that we already use, or defined these global rules, but I really want to add more static analysis to facilitate reviews and enhance code quality.

I will continue to look for the best way to do that.

@ferhatelmas Please don't fix all error shallow declarations :wink:

**Comment by timoreimann** [comment]
@ldez I still disagree: Only because there's a linter out there doesn't mean we need to wrestle our code in shape so that it pleases that linter. Especially in Go where there are tons of linters, and many of them work on a best-effort basis.

I don't find the `if` example referenced particularly tricky: once you know that if statements span a new scope, it's easy to deal with it (and, in fact, sometimes helpful). Adding new error variable name prefixes/suffixes each time is a worse option to me.

Linters are supposed to help us; we should have the freedom to skip linters if we feel they do more harm than good.

**Comment by ldez** [comment]
I do not want to continue the debate here, I will revert the shallow declaration changes.
But shallow declarations can lead to issue and we must find a solution.

**Pull request merged**